java/jdbc/src/main/java/com/flipkart/vitess/util/charset/Collation.java

```diff
@@ -0,0 +1,33 @@
+package com.flipkart.vitess.util.charset;
+
+/**
+ * These classes were pulled from mysql-connector-java and simplified to just the parts supporting the statically available
+ * charsets
+ */
+class Collation {
+    public final int index;
+    public final String collationName;
+    public final int priority;
```

Comment:
I tried not to modify the 3 classes in the charset package too much. They were taken directly from the latest version of mysql-connector-j. If you'd like I can make some changes.